v3iofs/fs.py

```diff
@@ -56,58 +54,70 @@ def __init__(self, v3io_api=None, v3io_access_key=None, **kw):
             transport_kind='requests',
         )
 
-    def ls(self, path, detail=True, **kwargs):
+    @classmethod
+    def _strip_protocol(cls, path):
+        """ Turn path from fully-qualified to file-system-specific
+        May require FS-specific handling, e.g., for relative paths or links.
+        """
+        path = stringify_path(path)
+        protos = (cls.protocol,) if isinstance(cls.protocol, str) else cls.protocol # NOQA
```

Comment:
What does this do?

Reply:
Checks to verify that a path is passed as a string.  If its passed as a filesystem object, attempts to convert it to a string.  Also identifies the protocol.  In this case, v3io